VideoWeb/VideoWeb/Controllers/ProfilesController.cs

```diff
@@ -29,20 +36,26 @@ public ProfilesController(IUserApiClient userApiClient, ILogger<ProfilesControll
         /// <returns></returns>
         [HttpGet]
         [SwaggerOperation(OperationId = "GetUserProfile")]
-        [ProducesResponseType(typeof(UserProfileResponse), (int)HttpStatusCode.OK)]
-        public async Task<IActionResult> GetUserProfileAsync()
+        [ProducesResponseType(typeof(UserProfileResponse), (int) HttpStatusCode.OK)]
+        public IActionResult GetUserProfile()
         {
-            var username = User.Identity.Name.ToLower().Trim();
             try
             {
-                var profile = await _userApiClient.GetUserByAdUserNameAsync(username);
-                var response = UserProfileResponseMapper.MapToResponseModel(profile);
+                var role = User.Claims.First(c => c.Type == ClaimTypes.Role).Value;
+                var response = new UserProfileResponse
+                {
+                    FirstName = User.Claims.First(c => c.Type == ClaimTypes.GivenName).Value,
```

Comment:
Can throw exceptions

Reply:
If we do this then we need to make a few more  changes to the the web app to assume a 500 response code means user is not authorised. Think that might be beyond the scope of this but I don't  mind doing it. Just worried about the domino effect